UCTNode field reorganization to optimize memory layout (attempt 2)

ihavnoid opened the pull request:
We had the discussion on how to reduce memory footprint and have many good ideas, but this is a simple stopgap change which is better than nothing until we have something better.

(prior discussion : https://github.com/gcp/leela-zero/pull/567)

72 bytes -> 56 bytes, this is 2 cache lines vs. 1 cache line in many CPUs.

gcp commented on src/UCTNode.h:
```diff
@@ -79,16 +79,11 @@ class UCTNode {
-
     // Move
-    int m_move;
+    int16_t m_move;
```
Needs a std:: prefix as it's not guaranteed to be in the global namespace. Also perhaps the move type should be a typedef if we do this.

gcp commented on src/UCTNode.h:
```diff
@@ -79,16 +79,11 @@ class UCTNode {
-    int m_move;
+    int16_t m_move;
     // UCT
+    std::atomic<int16_t> m_virtual_loss{0};
```
Same remark. Note that this could be an (u)int8_t too.

ihavnoid commented:
Pushed a new commit addressing concerns.  I did leave the m_virtual_loss at uint16_t simply because further reducing it won't save any memory footprint at this point.

gcp commented:
I guess in theory you could have 128 CPUs * 3 > 256 virtual losses anyway.
